docs/maintainers-guide.md

```diff
@@ -44,6 +45,9 @@
 - `README.md`: PineAPPL's repository README file. This file also serves as the
   homepage of [PineAPPL's website]
 - `_config.yml`: configuration file for PineAPPL's Github-pages website
+- `flake.lock`: Nix [flake lock file]
+- `flake.nix`: Nix [flake], which tracks cross-language dependencies to define a
+  reproducible build for the PineAPPL packages, and a suitable development shell
```

Comment:
The description is slightly beyond what is currently provided, since I have no `output.packages` in the flake (yet).

However, that is perfectly possible, and I'm also going to do it (sooner or later).
In principle, we could use Nix to build packages even in the CI (those for which we're distributing binaries). But dynamically-linked binaries won't be portable outside Nix (not sure if it's possible to make them such).

In any case, even spelling out the build is already a source distribution on its own, that will make it possible to compile PineAPPL with just:
```sh
nix build github:NNPDF/pineappl
```
and no extra requirement (not regarding build tools installation, nor dependencies).